runtime/src/main/java/org/corfudb/runtime/collections/CorfuStoreMetrics.java

```diff
@@ -0,0 +1,72 @@
+package org.corfudb.runtime.collections;
+
+import com.google.gson.JsonObject;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.MeterRegistry;
+import org.corfudb.common.metrics.micrometer.MeterRegistryProvider;
+
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class CorfuStoreMetrics {
+    // Keep track of the number of read entries to compute a stream's highestSequenceNumber
+    private final Optional<AtomicInteger> numberBatchReads;
+    private final Optional<AtomicInteger> numberReads;
```

Comment:
The type of these two metrics is Integer, which means they will be reset every time when getHighestSequenceNumber gets called. They won't have statistical data, is this what we want?

Reply:
I'm not on expert on this framework. But for these two metrics I actually used a gauge instead of a counter as according to the documentation:
"While a counter must always go up, gauges have no such constraint: a gauge outputs a value, unrelated from one call to the next." which is the case of these variable which are not increasing but are independent one from the other. Still I think this will give statistical data or are you sure it won't?

Reply:
I am pretty sure it won't give statistical data. 

> Gauge. A gauge is a metric that represents a single numerical value that can arbitrarily go up and down. Gauges are typically used for measured values like temperatures or current memory usage

Previously we use gauge to capture cache hit ratio or miss ratio, which is a single value when you access it. In this case, I think you can use histogram

Reply:
Yep you're correct, I looked into the metrics we collect and effectively there is no statistical data on these. In that case I think I'll use a DistributionSummary.